### PR TITLE
fix negative length in completion event data

### PR DESCRIPTION
**Pull request by yonip23**

### CODE REVIEW

File: src/selectionHandler.ts [hunk 1]
```diff
@@ -135,16 +140,18 @@ function eventDataOf(
   });
 
   const { length } = currentCompletion;
-  const netLength = editor.selection.anchor.character - position.character;
+  const netLength = length - (oldPrefix?.length || 0);
```
Comment (yonip23) [inline src/selectionHandler.ts:143]: this is the main change - `netLength` was calculated with positions, which is problematic if the completion takes more than one line. i changed it to be based on the completion parameters themselves (`old_prefix` is quite exactly the opposite of what we mean by `netLength`)

File: src/inlineSuggestions/inlineSuggestionState.ts [hunk 2]
```diff
@@ -52,6 +52,13 @@ export function getAllSuggestions(): ResultEntry[] | undefined {
   return results();
 }
 
+export function currentAutocompleteResult():
```
Comment (yonip23) [inline src/inlineSuggestions/inlineSuggestionState.ts:55]: remove

File: src/selectionHandler.ts (hunk 1 above)
Comment (yonip23) [reply src/selectionHandler.ts:143]: this is also how we calculate it in the intellij plugin: https://github.com/codota/tabnine-intellij/blob/5c66c60f08d60463cf87b372264b3638c6faee1a/src/main/java/com/tabnine/selections/CompletionPreviewListener.java#L27

**Review by dimacodota — APPROVED** [review]
lgtm

**Comment by dimacodota** [comment]
add some tests

**Pull request merged**